### ISSUE TITLE
W1307 false positive with {N[identifier]} format specifiers

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#288]
Originally reported by: **Chris Rebert (BitBucket: [cvrebert](http://bitbucket.org/cvrebert), GitHub: @cvrebert?)**

---

```
#!bash

$ pylint --version
No config file found, using default configuration
pylint 1.3.0, 
astroid 1.2.0, common 0.62.0
Python 2.7.7 (default, Jun 14 2014, 23:12:13) 
[GCC 4.2.1 Compatible Apple LLVM 5.1 (clang-503.0.40)]
```

```
#!python

# foobarbaz.py 
def greet(**kwargs):
    return 'Hello {0[name]}'.format(kwargs)
```

```
#!bash

$ pylint -r n -f parseable foobarbaz.py 
************* Module foobarbaz
foobarbaz.py:2: [W1307(invalid-format-index), greet] Using invalid lookup key 'name' in format specifier "0['name']"
```

However, despite the warning, the code seems to work just fine:

```
#!python

Python 2.7.7 (default, Jun 14 2014, 23:12:13) 
[GCC 4.2.1 Compatible Apple LLVM 5.1 (clang-503.0.40)] on darwin
>>> from foobarbaz import greet
>>> greet(name=42)
'Hello 42'
>>> 
```

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/288
